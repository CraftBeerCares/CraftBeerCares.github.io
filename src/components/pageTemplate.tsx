import styles from "./pageTemplate.module.css";
import Footer from "./footer";
import Header from "./header";

export default function PageTemplate({
  children,
}: {
  children: React.ReactNode;
}) {
  return (
    <div className="container">
      <Header />
      <main className={styles.mainContent}>{children}</main>
      <Footer />
    </div>
  );
}
