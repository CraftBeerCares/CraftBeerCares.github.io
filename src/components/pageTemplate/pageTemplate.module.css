.pageContainer {
  width: 100%;
  height: 100vh;
  display: flex;
  flex-direction: column;
}

.header {
  position: sticky;
  top: 0;
  border-bottom: 3px solid var(--cbc-pink);
}

.mainContent {
  flex-grow: 1;
  max-width: var(--max-width);
  margin-left: auto;
  margin-right: auto;
}

.contentContainer {
  overflow-y: auto;
}

/* Mobile */
@media (max-width: 700px) {
}

/* Tablet Smaller Desktop */
@media (min-width: 701px) {
}
