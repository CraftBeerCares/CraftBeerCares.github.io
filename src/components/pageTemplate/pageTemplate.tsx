import styles from "./pageTemplate.module.css";
import { Footer, Header } from "@/components";
import { primary } from "@/styles/fonts";
import { clsx } from "clsx";

interface PageTemplateProps {
  children: React.ReactNode;
}

export const PageTemplate = ({ children }: PageTemplateProps) => {
  return (
    <div className={styles.pageContainer}>
      <div className={styles.header}>
        <Header />
      </div>
      <div className={styles.contentContainer}>
        <main className={clsx(primary.className, styles.mainContent)}>
          {children}
        </main>
        <Footer />
      </div>
    </div>
  );
};
