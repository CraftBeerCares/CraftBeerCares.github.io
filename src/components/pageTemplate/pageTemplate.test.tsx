import { render } from "@testing-library/react";
import {PageTemplate} from "./pageTemplate";

it("renders Page Template", () => {
  const { container } = render(<PageTemplate>Hello</PageTemplate>);
  expect(container).toMatchSnapshot();
});
