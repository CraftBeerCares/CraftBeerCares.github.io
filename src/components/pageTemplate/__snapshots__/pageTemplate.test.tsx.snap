// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Page Template 1`] = `
<div>
  <div
    class="pageContainer"
  >
    <div
      class="header"
    >
      <div
        class="content"
      >
        <header>
          <div
            class="header"
          >
            <button
              class="burgerMenuButton"
            >
              <img
                alt="Burger Menu"
                data-nimg="fill"
                decoding="async"
                loading="lazy"
                src="/assets/menu.svg"
                style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; object-fit: contain; color: transparent; filter: invert(1);"
              />
            </button>
            <h1
              class="className title"
            >
              Craft Beer Cares
            </h1>
            <nav
              class="topNav className"
            >
              <a
                class="navLink"
                href="/"
              >
                Home
              </a>
              <a
                class="navLink"
                href="/about"
              >
                About
              </a>
              <a
                class="navLink"
                href="/join_us"
              >
                Join Us
              </a>
              <a
                class="navLink"
                href="/events"
              >
                Events
              </a>
              <a
                class="navLink"
                href="/shop"
              >
                Shop
              </a>
              <a
                class="navLink"
                href="/contact"
              >
                Contact
              </a>
            </nav>
          </div>
        </header>
      </div>
    </div>
    <div
      class="scrollContainer"
    >
      <main
        class="className content"
      >
        Hello
      </main>
      <div
        class="footer"
      >
        <div
          class="content"
        >
          <footer
            class="footer"
          >
            <div
              class="footerMobile"
            >
              <div
                class="footerLogo"
              >
                <img
                  alt="Craft Beer Cares Logo"
                  data-nimg="fill"
                  decoding="async"
                  loading="lazy"
                  src="/assets/logo_bw.svg"
                  style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; filter: invert(1);"
                />
              </div>
            </div>
            <div
              class="footerDesktop"
            >
              <div
                class="footerColumn"
              >
                <h4
                  class="className title"
                >
                  Craft Beer Cares
                </h4>
                <nav
                  class="className footerLinks"
                >
                  <a
                    class="navLink"
                    href="/"
                  >
                    Home
                  </a>
                  <a
                    class="navLink"
                    href="/about"
                  >
                    About
                  </a>
                  <a
                    class="navLink"
                    href="/join_us"
                  >
                    Join Us
                  </a>
                  <a
                    class="navLink"
                    href="/events"
                  >
                    Events
                  </a>
                  <a
                    class="navLink"
                    href="/shop"
                  >
                    Shop
                  </a>
                  <a
                    class="navLink"
                    href="/contact"
                  >
                    Contact
                  </a>
                </nav>
                <div
                  class="socialLinks"
                >
                  <a
                    class="socialLink"
                    href="https://www.facebook.com/craftbeercares"
                  >
                    <img
                      alt="Facebook"
                      data-nimg="1"
                      decoding="async"
                      height="30"
                      loading="lazy"
                      src="/assets/facebook.svg"
                      style="color: transparent;"
                      width="30"
                    />
                  </a>
                  <a
                    class="socialLink"
                    href="https://www.instagram.com/craftbeercares/"
                  >
                    <img
                      alt="Instagram"
                      data-nimg="1"
                      decoding="async"
                      height="30"
                      loading="lazy"
                      src="/assets/instagram.svg"
                      style="color: transparent;"
                      width="30"
                    />
                  </a>
                  <a
                    class="socialLink"
                    href="https://twitter.com/craftbeercares"
                  >
                    <img
                      alt="Twitter"
                      data-nimg="1"
                      decoding="async"
                      height="30"
                      loading="lazy"
                      src="/assets/twitter.svg"
                      style="color: transparent;"
                      width="30"
                    />
                  </a>
                </div>
                <div
                  class="className footerQuote"
                >
                  Craft Beer Cares - Drinking beer for good Stuff since 2017
                </div>
              </div>
              <div
                class="footerColumn"
              >
                <div
                  class="footerLogo"
                >
                  <img
                    alt="Craft Beer Cares Logo"
                    data-nimg="fill"
                    decoding="async"
                    loading="lazy"
                    src="/assets/logo_bw.svg"
                    style="position: absolute; height: 100%; width: 100%; left: 0px; top: 0px; right: 0px; bottom: 0px; color: transparent; filter: invert(1);"
                  />
                </div>
              </div>
              <div
                class="footerColumn rightColumn"
              >
                <div
                  class="signUpBox"
                >
                  <h4
                    class="className title"
                  >
                    Sign Up for Our Monthlyish Newsletters
                  </h4>
                  <p
                    class="className signUpText"
                  >
                    Hear about upcoming events and initiatives from us and others pairing beer with charitable giving. We wonâ€™t share your info with anyone, and you can opt out at anytime.
                  </p>
                  <input
                    class="emailBox"
                    placeholder="Your e-mail"
                  />
                  <button
                    class="signUpButton"
                  >
                    Sign Up!
                  </button>
                </div>
              </div>
            </div>
          </footer>
        </div>
      </div>
    </div>
  </div>
</div>
`;
